class-20.md

```diff
@@ -0,0 +1,48 @@
+
+### Tendermint
+
+Tendermint is an implementation of a dBFT blockchain. It can use a cryptocurrency optionally, or be used simply as a database. You can check a very popular implemenation of it in the Smart Cosmos network. A rough diagram of the implementation can be seen below:
+
+![Tendermint Consensus](https://tendermint.com/docs/assets/img/tm-transaction-flow.258ca020.png)
+
+Blocks go through a series of pre-processing and eventually commit transactions to a block. The rest of the blockchain structure remains intact, but the only parties that have write permission on blocks can produce new data in them. A block is proposed in the process:
+
+1. Block proposer candidate for the current round is selected
+2. Block proposer sends out a block holding the proposed transactions
+3. Each node indicates that it is ready to participate in the voting process
+4. Each node votes to accept the block (need 2/3 of votes to pass the voting round)
+5. If all validators vote yes, a new block is broadcast and added to the chain
+
+This process allows block producers to circulate large blocks quickly, and not worry about checking approval outside of validators quickly. While bitcoin (at lecture time) processes far under 10 transactions per second and Ethereum under 20, DPOS systems promise thousands or tens of thousands of transactions per second, rivaling modern credit card processing networks in instances.
+
+##### Fault Tolerance
+
+These systems could be succeptible to byzantine faults, but in most cases it would take a very large amount of bad actors on the network. It would take over 1/3 of nodes acting in bad faith to cause a byzantine fault in a dBFT network to cause an error due to a byzantine fault.
+
+### DPOS
+
+Building on a tendermint-based system, validator nodes can be elected based on a comination of:
+
+1. Amount of cryptocurrency staked
+2. Amount of votes received from the community
+
+Through a combination of financial incentives to continue producing blocks and earning block rewards, block proposer or witness nodes have a financial incentive to continue to act in good faith.
+
+A set of dBFT blockchains promoted by Dan Larimer following an architecture called Graphene follows this. The first, Bitshares, allows for simple defining of assets and a decentralized exchange with listing/making/taking orders. This evolved first to Steem - a social network built on the blockchain. Instead of sending simple transactions, posts in social media were sent, and likes/comments were sent onto the blockchain as operations. The higher throughput of dBFT blockchains enabled this (all of a social network sharing 20 transactions per second wouldn't realistically make sense). 
```

Comment:
`A set of dBFT blockchains promoted by Dan Larimer following an architecture called Graphene follows this.`
-> Says 'follows' twice here

`The first, Bitshares, allows for simple defining of assets and a decentralized exchange with listing/making/taking orders.`
-> `allows for simple definitions of assets and decentralized exchange with creating/listing/taking orders` (?)
-> Doesn't really make sense. Is the exchange that's "with creating/listing/taking orders", or is it simple definitions that's "with creating/listing/taking orders"?

Reply:
`(all of a social network sharing 20 transactions per second wouldn't realistically make sense).`
-> Why doesn't it make sense? Need to elaborate.